orca-front50/src/main/groovy/com/netflix/spinnaker/orca/front50/tasks/DeliveryConfigUtils.java

```diff
@@ -0,0 +1,51 @@
+package com.netflix.spinnaker.orca.front50.tasks;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.netflix.spinnaker.orca.front50.Front50Service;
+import com.netflix.spinnaker.orca.front50.model.DeliveryConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import retrofit.RetrofitError;
+import retrofit.client.Response;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Optional;
+
+public class DeliveryConfigUtils {
+  private Logger log = LoggerFactory.getLogger(getClass());
+  private Front50Service front50Service;
+  private ObjectMapper objectMapper;
+
+  public DeliveryConfigUtils(Front50Service front50Service, ObjectMapper objectMapper) {
+    this.front50Service = front50Service;
+    this.objectMapper = objectMapper;
+  }
+
+  public Optional<DeliveryConfig> getDeliveryConfig(String id) {
+    try {
+      DeliveryConfig deliveryConfig = readResponse(front50Service.getDeliveryConfig(id));
+      return deliveryConfig == null ? Optional.empty() : Optional.of(deliveryConfig);
+    } catch (RetrofitError e) {
+      //ignore an unknown (404) or unauthorized (403)
+      if (e.getResponse() != null && Arrays.asList(404, 403).contains(e.getResponse().getStatus())) {
+        return Optional.empty();
+      } else {
+        throw e;
+      }
+    }
+  }
+
+  public DeliveryConfig readResponse(Response response) {
+    try {
+      return objectMapper.readValue(response.getBody().in(), DeliveryConfig.class);
+    } catch (IOException e) {
+      log.error("Unable to deserialize delivery config, reason: ", e.getMessage());
```

Comment:
If this fails in production, we are back to the same problem of not being able to diagnose the root cause without cracking open logs. Would be nicer if the exception ended up in the context somehow

Reply:
Could you explain this more? Is your desired state being able to diagnose everything by looking at the execution context and not the logs? Are there any errors that are best only in the logs? Any ideas on how to surface something like this in the execution context?

Reply:
I'd prefer this to b `log.error("Unable to deserialize delivery config", e)` -- then we'll get the full stacktrace.

Reply:
I think, ideally, we surface the relevant errors to the end user so they can self-service issues. Thinking about error messages... but the immediate response is based on several recent support issues where folks ask us why a pipeline fails and we have to go to logs to see the exception and tell them what it said. Ideally we short circuit that. Will get back when I have some more thoughts about it